api/android/api/src/main/jni/nnstreamer-native-api.c

```diff
@@ -463,16 +461,15 @@ nns_parse_tensors_data (pipeline_info_s * pipe_info, JNIEnv * env,
       gsize data_size = (gsize) (*env)->GetDirectBufferCapacity (env, tensor);
       gpointer data_ptr = (*env)->GetDirectBufferAddress (env, tensor);
 
-      if (data->tensors[i].tensor == NULL)
-        data->tensors[i].tensor = g_malloc (data_size);
-      if (data->tensors[i].tensor == NULL) {
-        nns_loge ("Failed to allocate memory %zd, index %d.", data_size, i);
-        (*env)->DeleteLocalRef (env, tensor);
-        failed = TRUE;
-        goto done;
+      if (clone) {
+        if (data->tensors[i].tensor == NULL)
+          data->tensors[i].tensor = g_malloc (data_size);
+
+        memcpy (data->tensors[i].tensor, data_ptr, data_size);
```

Comment:
Need to handle memory error case here?

Reply:
low mem case, g_malloc will break if failed to allocate mem size :)

Reply:
Oh right!